honeybee/room.py

```diff
@@ -0,0 +1,605 @@
+                allowed to differ from one another in order to consider them colinear.
+
+        Usage:
+            office_room = Room.from_faces('My Office', faces, 0.001, 1)
+            assert office_room.geometry.is_solid, 'faces do not form a closed volume!'
+            print(office_room.volume)
+        """
+        # check the input faces
+        for face in faces:
+            assert isinstance(face, Face), \
+                'Expected honeybee Face. Got {}'.format(type(face))
+        face_geometries = tuple(face.geometry for face in faces)
+
+        # try to get a closed volume between the faces
+        if tolerance == 0:
+            room_polyface = Polyface3D.from_faces(face_geometries)
+        else:
+            room_polyface = Polyface3D.from_faces_tolerance(face_geometries, tolerance)
+        if not room_polyface.is_solid:
+            ang_tol = math.radians(angle_tolerance)
+            room_polyface = room_polyface.merge_overlapping_edges(tolerance, ang_tol)
+
+        # replace honeybee face geometry with versions that are facing outwards
+        if room_polyface.is_solid:
+            for i, correct_face3d in enumerate(room_polyface.faces):
+                faces[i]._geometry = correct_face3d
+        room = cls(name, faces)
+        room._geometry = room_polyface
+        return room
+
+    @classmethod
+    def from_polyface3d(cls, name, polyface, roof_angle=30, floor_angle=150,
+                        ground_depth=0):
+        """Initialize a Room from a ladybug_geometry Polyface3D object.
+
+        Args:
+            name: Room name. Must be < 100 characters.
+            polyface: A ladybug_geometry Polyface3D object representing the closed
+                volume of a room. The Polyface3D.is_solid property can be used to
+                determine whether the polyface is a closed solid before input here.
+            roof_angle: Cutting angle for roof from Z axis in degrees. Default: 30.
+            floor_angle: Cutting angle for floor from Z axis in degrees. Default: 150.
+            ground_depth: The Z value above which faces are considered Outdoors
+                instead of Ground. Faces will have a Ground boundary condition if
+                their center point lies at or below this value. Default: 0.
+        """
+        assert isinstance(polyface, Polyface3D), \
+            'Expected ladybug_geometry Polyface3D. Got {}'.format(type(polyface))
+        faces = []
+        for i, face in enumerate(polyface.faces):
+            faces.append(Face('{}_Face{}'.format(name, i), face,
+                              get_type_from_normal(face.normal, roof_angle, floor_angle),
+                              get_bc_from_position(face.center, ground_depth)))
+        room = cls(name, faces)
+        room._geometry = polyface
+        return room
+
+    @classmethod
+    def from_box(cls, name, width=3.0, depth=6.0, height=3.2,
+                 orientation_angle=0, origin=Point3D(0, 0, 0)):
```

Comment:
I know this is a left-over from honeybee room but `rotation_angle` might be a better choice than `orientation_angle`.

Reply:
Actually, `rotation_angle` was what you originally had for the older honeybee Room but I thought it didn't provide any clarity as to whether the angle was like a standard rotation operation, which typically follows right-hand rule logic to go counterclockwise, or whether it was trying to match a specific cardinal orientation, which typically starts from the north and goes clockwise.  I looked at your code ans it was the latter so I thought `orientation_angle` was clearer.  If there was some logic to your original choice of the word `rotation_angle` that was lost on me, please elaborate.

Reply:
To me both terms are very similar. Don't have a strong opinion. If this is closer to what the angle does then let's keep it as is.